src/corelibs/U2Core/src/gobjects/MultipleChromatogramAlignmentObject.cpp

```diff
@@ -260,6 +260,47 @@ void MultipleChromatogramAlignmentObject::trimRow(const int rowIndex, int curren
     updateCachedMultipleAlignment(modificationInfo);
 }
 
+void MultipleChromatogramAlignmentObject::updateAlternativeMutations(const McaReadsTabSettings& settings, U2OpStatus& os) {
+    for (int i = 0; i < getNumRows(); i++) {
+        const MultipleChromatogramAlignmentRow& mcaRow = static_cast<const MultipleChromatogramAlignmentRow&>(getRow(i));
+        const int ungappedLength = mcaRow->getUngappedLength();
+
+        QMap<qint64, char> newCharList;
```

Comment:
Why map but not hash?

Reply:
Well.... OK, replaced, now this code executes 0.01 seconds faster)